lib/thread.c

```diff
@@ -1669,6 +1726,37 @@ void thread_call(struct thread *thread)
 	unsigned long helper;
 #endif
 	RUSAGE_T before, after;
+	struct timeval wkptime, diff;
+	uint64_t wkpintvl_usecs;
+
+	/*
+	 * The following code measures interval between consecutive
+	 * wakeup of the various threads under the current process
+	 * and log that in order to detect any possible CPU starvations.
+	 * Offcourse it can not be used to differentiate between
+	 * wether a process was indeed ready but it could not be
+	 * scheduled (due to other process hogging CPU) and that of
+	 * the process really being idle for long.
+	 *
+	 * In cases if a deamon is supposed to run some timely regular
+	 * jobs, the data captured here can definitely highlight if
+	 * the process ever went through CPU starvation or not.
+	 */
+	frr_get_realtime(&wkptime);
```

Comment:
how doesn't GETRUSAGE provide this data?  Why do we need more time gathering when we are already doing it?

Reply:
Like mentioned before the monotime_xxx() functions record monotonic time that cannot be used to display real clock time with microseconds granularity. This timestamp is being recorded for both computing interval as well as recording timestamps with milliseconds granularity.

Reply:
I would point out that we use monotime *extensively* for thread cpu timekeeping.  Why is your use case any different here?

Reply:
The need is not only to record time-interval (between current wakeup time and the last wakeup time) but is also to record the timestamp like 2021-01-22 07:20:39.522846 for the last wakeup time for display as well. Timestamps taken with monotime() cannot be displayed in a calendar like format. At least I could not get a proper time display string with microseconds information with the existing monotime_xxxx() functions.

Reply:
Removed usage of frr_get_realtime().